rfcs/0100-sign-commits.md

```diff
@@ -0,0 +1,345 @@
+by passing filters (see man page for `git-clone(1)` and `git-rev-list(1)`):
+- `git clone git://github.com/NixOS/nixpkgs --single-branch --branch=21.05-beta` 4:17
+- `git clone git://github.com/NixOS/nixpkgs --single-branch --branch=21.05-beta --filter=blob:none` 1:05
+- `git clone git://github.com/NixOS/nixpkgs --single-branch --branch=21.05-beta --filter=tree:0` 0:22
+- `git clone git://github.com/NixOS/nixpkgs --single-branch --branch=21.05-beta --depth 1` 0:08
+  (Almost supported already, see https://github.com/NixOS/nix/issues/5119)
+- `curl https://github.com/NixOS/nixpkgs/archive/21.05-beta.tar.gz -LO | tar -xf -`
+  0:04 (roughly equivalent to `github:NixOS/nixpkgs?ref=21.05-beta`)
+
+Then after doing `cd nixpkgs`, we try fetching `21.05-beta`, which is 659 commits more to
+pull:
+- `git pull --ff-only origin 21.05` 0:08
+- `git pull --ff-only origin 21.05` 0:08 (after having fetched with `--filter=blob:none`)
+- `git pull --ff-only origin 21.05` 0:07 (after having fetched with `--filter=tree:0`)
+- `git clone git://github.com/NixOS/nixpkgs --single-branch --branch=21.05 --depth 1` 0:08 (pulling doesn't work)
+- `curl https://github.com/NixOS/nixpkgs/archive/21.05.tar.gz -LO | tar -xf -`
+  0:04 (roughly equivalent to `github:NixOS/nixpkgs?ref=21.05`)
+
+This is with an 850 EVO SSD, the average download speed from GitHub was around 11 MiB/s according to `git`.
+Relevant documentation is:
+- https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/
+- https://git-scm.com/docs/partial-clone
+- https://git-scm.com/docs/shallow
+
+We likely want to make Nix use `--filter=tree:0` by default when fetching repositories.
+`--depth 1` is faster, but it doesn't preserve the commit history,
+and it's not clear how it it interacts with PGP signing.
+
+# Examples and Interactions
+[examples-and-interactions]: #examples-and-interactions
+
+Committers will need to adapt to the new workflow before March of 2022.
+The command they will run is explained in [signing], but further
+information about how to sign commits is out of scope for this RFC.
+
+Relevant documentation for setting up GPG:
+- https://guix.gnu.org/manual/en/html_node/Commit-Access.html
+- https://www.kernel.org/doc/html/latest/process/maintainer-pgp-guide.html
+
+Flake users will need to become aware of the basics of this system
+when using forks of Nixpkgs, as updating such flake inputs
+will be different from before.
+In addition, flake users should also begin taking use of the features
+introduced in this RFC, as they are usable and useful for all flakes.
+
+## Example flake
+
+```nix
+# flake.nix
+{
+  inputs.someflake = {
+    url = "https://github.com/some/flake.git";
+    startsAt = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
+    startsAtKey = "XXXX XXXX XXXX XXXX XXXX  XXXX XXXX XXXX XXXX XXXX";
+    type = "git";
+  };
+
+  outputs = { someflake }: {};
+}
+```
+
+```nix
+# some/flake/authorized-committers.nix
```

Comment:
On the matter of where to put those files: Wasn't there some standard to put them into a `.well-known` folder? It hasn't really gained traction but IMO a better place than spewing them into the root of a repository.

Reply:
I chose something that mirrors the filename Guix uses, however, I think that's better, thanks.